WordPress/Classes/Models/Post.m

```diff
@@ -196,18 +196,19 @@ - (NSString *)contentPreviewForDisplay
 
 - (NSString *)statusForDisplay
 {
-    NSString *statusString = [self statusTitle];
-    if ([self.status isEqualToString:PostStatusPublish] || [self.status isEqualToString:PostStatusDraft]) {
-        statusString = nil;
-    }
-    if (![self hasRemote]) {
+    NSString *statusString = nil;
+    
+    if (self.isRevision) {
         NSString *localOnly = NSLocalizedString(@"Local", @"A status label for a post that only exists on the user's iOS device, and has not yet been published to their blog.");
         if (statusString) {
             statusString = [NSString stringWithFormat:@"%@, %@", statusString, localOnly];
         } else {
             statusString = localOnly;
```

Comment:
> The way this is composed, statusString will always be nil for a revision thus only the else clause will be executed. 

The idea behind this PR is that only non-uploaded changes are marked as "revisions".  So whenever something is marked as a revision, it means there are local changes that were never uploaded to the server.

> This is problematic for scheduled posts, and posts that are pending review as we would still want to display those statuses.

I'm testing it in the simulator here and it's working as I'd expect it to.  Some example cases:

1. Post is immediate, but failed to upload: "Local"
2. Post is immediate, and uploaded successfully: nil.
3. Post is scheduled, but failed to upload: marked "Local"
4. Post is scheduled, and uploaded successfully: "Scheduled"
5. Post is awaiting review, but failed to upload: "Local"
6. Post is awaiting review, and uploaded successfully: "Awaiting review"

The idea of the "Local" tag is to let the user know he will potentially loose those changes if he doesnt try to upload them again.